Fix docs build

felixfontein opened the pull request:
https://github.com/ansible/ansible/pull/85095 broke the docs build since `AnsibleModule`'s documentation now references a class `PluginInfo`, which isn't currently documented on the docsite.

The error message is:
```
docs/docsite/rst/index.rst:0:0: unknown: /tmp/docs-build-ndrdw7z9-sanity/lib/ansible/module_utils/basic.py:docstring of ansible.module_utils.basic.AnsibleModule.deprecate:1: WARNING: py:class reference target not found: ansible.module_utils.common.messages.PluginInfo
```

felixfontein committed:
Document PluginInfo.

felixfontein commented:
So this works. The main question is where `PluginInfo` should be documented. I've currently added it below `AnsibleModule`. Maybe further down is better?

oraNod commented:
> So this works. The main question is where `PluginInfo` should be documented. I've currently added it below `AnsibleModule`. Maybe further down is better?

I feel like it should go between basic and arg spec. Maybe since `PluginInfo` is provisional it should go after arg spec, but maybe a bit higher makes it more prominent?

In any case I feel like it's good to keep things in `ansible.module_utils.basic` and `ansible.module_utils.common` grouped together.

felixfontein committed:
Move PluginInfo before Argument Spec, and fix heading underlines.

felixfontein commented:
Sounds like a good idea. I've adjusted the PR accordingly, and also fixed the headling underlines to conform to https://docs.ansible.com/ansible/devel/dev_guide/style_guide/#heading-notation.

samccann reviewed (approved):
LGTM and thanks for the fix!

felixfontein commented:
@oraNod @samccann thanks for reviewing and merging!
